include/zcbor_decode.h

```diff
@@ -146,6 +150,92 @@ bool zcbor_uint64_expect_union(zcbor_state_t *state, uint64_t expected);
  */
 bool zcbor_list_start_decode(zcbor_state_t *state);
 bool zcbor_map_start_decode(zcbor_state_t *state);
+bool zcbor_unordered_map_start_decode(zcbor_state_t *state);
+
+/** Search for a key in a map.
+ *
+ * By spec, elements (key-value pairs) in CBOR maps can appear in any order.
+ * Use this function when you don't know the order of the elements of a map.
+ *
+ * This must only be used while inside a map that has been entered via
+ * @ref zcbor_unordered_map_start_decode. Use @ref zcbor_unordered_map_end_decode
+ * when leaving the map.
+ *
+ * When this function has returned, it can never immediately be called again
```

Comment:
`never immediately` sounds a bit weird, how about `it cannot be immediately called again` or simply just state what needs to be done instead. Its easier to explain what a user must do than what they must not do.

Reply:
Thanks, I rephrased this part.